Port debug commands to RzShell

Rot127 opened the pull request:
<!-- Filling this template is mandatory -->

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

<!-- Explain the **details** for making this change. Is a new feature implemented? What existing problem does the pull request solve? How does the pull request solve these issues? Please provide enough information so that others can review your pull request. -->

- Remove `dmiv` command, since it is an alias for `dmi`
- Removes `dta` command:
  Originally it limited the addresses to trace.
  But it was not used in any test, and the implementation was not optimal.
  It allowed only specific addresses (no range).
  Also worked on strings (not ut64 values).
  Also, the usefulness is limited, considering the alternative
  to just not tracing a certain address.
- Removes undocumented `dmias` command, because no idea what it does.

**Test plan**

<!-- What steps should the reviewer take to test your pull request? Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots/videos. This is your time to re-check that everything works and that you covered all the edge cases -->

...

**Closing issues**

<!-- put "closes #XXXX" in your comment to auto-close the issue that your PR fixes (if any). -->

...

Rot127 committed:
Port dmhb to RzShell.

Rot127 committed:
Port dmhf to RzShell

Rot127 committed:
Remove 'to' argument from rz_core_debug_continue_until().

It was never used in the whole function.
Hence, whatever it was supposed to do, it didn't

Rot127 committed:
Port dcu to RzShell

Rot127 committed:
Fix tests of dmh[bf] commands

Rot127 committed:
Port dmi commands to RzShell

Rot127 committed:
Port dmx to RzShell

Rot127 committed:
Format code

Rot127 committed:
Fix typo dmw -> dmx

Rot127 committed:
Remove dta command.

Originally it limited the addresses to trace.
But it was not used in any test, and the implementation was not optimal.
It allowed only specific addresses (no range).
Also worked on strings (not ut64 values).
Also, the usefulness is limited, considering the alternative
to just not tracing a certain address.

XVilka commented on librz/core/cdebug.c:
```diff
@@ -96,13 +96,25 @@ RZ_IPI void rz_core_debug_continue(RzCore *core) {
+ * \param addr The address to execute to.
+ *
+ * \return true On success.
+ * \return false On otherwise.
```
Just "otherwise".

XVilka commented on librz/include/rz_cmd.h:
```diff
@@ -91,6 +91,7 @@ typedef enum {
 	RZ_OUTPUT_MODE_LONG_JSON = 1 << 6,
 	RZ_OUTPUT_MODE_TABLE = 1 << 7,
 	RZ_OUTPUT_MODE_QUIETEST = 1 << 8,
+	RZ_OUTPUT_MODE_GRAPH = 1 << 9,
```
Nice addition but there could be multiple types of graph - e.g. native, dot one, interactive, and so on. While it's less relevant in this PR, we should think this over nevertheless, possibly at some later stage. But preferably before the release.

Rot127 replied:
Yeah, would be nice. I can attempt to add it here. If it is too much struggle we can still do it later.

Rot127 replied:
Ok, wouldn't do it here. Because it requires some refactor. Not necessarily. We can just pass core in this case, but we shouldn't complicate our code more, if we need to modularize the graph implementation nonetheless.

See: https://github.com/rizinorg/rizin/issues/4231#issuecomment-2512159813

XVilka replied:
Sounds good to me.

Rot127 committed:
Fix unused variable warning in release build.

Rot127 committed:
Add quiet and quietest mode to dmi commands.

This also removes the test for dmias.
The command was not documented and it doesn't seem to
do anything special dmia can't do.

XVilka reviewed (approved):
@wargio take a look please

XVilka commented:
@Rot127 could you please resend this PR from inside rizinorg with a branch starting from `dist-` to trigger all CI jobs?

Rot127 committed:
Fix dawrin tests.

The debug log was printed via eprintf and now via RZ_LOG_INFO.
Hence it is gone.

wargio commented on librz/core/cdebug.c:
```diff
@@ -127,35 +138,37 @@ RZ_API bool rz_core_debug_continue_until(RzCore *core, ut64 addr, ut64 to) {
 				prev_ret = false;
 			}
 			if (steps % 500 == 0 || pc == addr) {
-				eprintf("At 0x%08" PFMT64x " after %lu steps\n", pc, steps);
+				RZ_LOG_DEBUG("At 0x%08" PFMT64x " after %lu steps\n", pc, steps);
 			}
```
optimize this since RZ_LOG_DEBUG is not compiled and this branch becomes dead code, by adding `#if RZ_BUILD_DEBUG`

wargio replied:
you can define a new logger within cdebug.c and have this clause which is used to print when x times has been printed.

Rot127 replied:
Doesn't the compiler remove it in this case?

wargio replied:
depends on the optimizations, i would still add the if/endif

Rot127 replied:
Done

wargio commented on librz/core/linux_heap_glibc.c:
```diff
@@ -1307,30 +1307,25 @@ static void GH(print_heap_bin)(RzCore *core, GHT m_arena, MallocState *main_aren
+	if (!input[0]) {
 		PRINT_YA("Bins {\n");
-		for (i = 0; i < NBINS - 1; i++) {
+		for (int i = 0; i < NBINS - 1; i++) {
```
i hope `NBINS` is a macro, otherwise is a terrible idea to have `signed int` here.

Rot127 replied:
It is `NBINS = 128`. Changed it to `size_t` nonetheless.

wargio reviewed (approved):
LGTM, beside that small comment

Rot127 committed:
Make i a size_t and fix string formatting

Rot127 committed:
Exclude debug log from release build.

wargio commented:
```
../librz/core/cdebug.c:115:17: error: variable 'steps' set but not used [-Werror,-Wunused-but-set-variable]
                unsigned long steps = 0;
```
